core/server/common/src/main/java/alluxio/master/StateLockOptions.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master;
+
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+
+import java.util.concurrent.locks.Lock;
+
+/**
+ * Used to define state-lock options for taking it exclusively.
+ * A typical framework for obtaining the state-lock exclusively is:
+ *  1- Run a grace-cycle:
+ *      cycle of {@link Lock#tryLock()} - {@link Thread#sleep(long)} calls.
+ *  2- Interrupt state-lock holders if grace-cycle fails.
```

Comment:
This is actually something like "continually interrupt state-lock holders and waiters", right?